drivers/rtc/rtc_rpi_pico.c

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright (c) 2020 Raspberry Pi (Trading) Ltd.
```

Comment:
Note to reviewers: I've added this copyright notice here and I'd be grateful if someone to check they're happy with this. https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/hardware_rtc/rtc.c#L147 is in part a source of some of the work here. I think this is ok from a licensing perspective, and given there's only a very small number of ways of setting bits in register.